crates/red_knot/src/main.rs

```diff
@@ -164,16 +164,8 @@ fn run() -> anyhow::Result<ExitStatus> {
 
     // TODO: Verify the remaining search path settings eagerly.
     let site_packages = venv_path
-        .map(|venv_path| {
-            let venv_path = SystemPath::absolute(venv_path, &cli_base_path);
-
-            if system.is_directory(&venv_path) {
-                Ok(site_packages_dirs_of_venv(&venv_path, &system)?)
-            } else {
-                Err(anyhow!(
-                    "Provided venv-path {venv_path} is not a directory!"
-                ))
-            }
+        .map(|path| {
+            VirtualEnvironment::new(path, &system).and_then(|venv| venv.site_packages_dirs(&system))
```

Comment:
I think it's important that the CLI converts the `path` because IMO, the path should be relative to the CLI's current working directory and not `--current-directory`.

E.g. when I run the following command in `/project/ruff`

```shell
red_knot --current-directory ../test/sub --virtual-env ../venv`  
```

I would expect that `../venv` resolves to `/project/venv` and `current-directory` to `/project/test/sub`. I would be surprised if virtual env resolves to `/project/test/venv`

Reply:
I wouldn't be entirely sure which of those results to expect, mostly because of the name `--current-directory` for that flag. I think your interpretation is fine, but it would be more obvious to me if instead of a `--current-directory` flag we just accepted the directory to check as an optional positional argument on the CLI, rather than a flag argument. (Also it's less to type :) )